test/core/templaters/jinja_test.py

```diff
@@ -411,6 +411,20 @@ def test__templater_jinja_error_variable():
     assert any(v.rule_code() == "TMP" and v.line_no == 1 for v in vs)
 
 
+def test__templater_jinja_dynamic_variable_no_violations():
+    """Test no templater violation for variable defined within template."""
+    t = JinjaTemplater(override_context=dict(blah="foo"))
+    instr = """{% if True %}
+    {% set some_var %}1{% endset %}
+    SELECT {{some_var}}
+{% endif %}
+"""
+    outstr, vs = t.process(in_str=instr, fname="test", config=FluffConfig())
+    assert str(outstr) == "\n    \n    SELECT 1\n\n"
+    # Check we have violations.
+    assert not len(vs) > 0
```

Comment:
Why not this?

```suggestion
    assert len(vs) == 0
```

Reply:
Oh yes, stupid mistake on my part -- I was in a hurry and not paying attention. Will change.